shell/browser/event_emitter_mixin.h

```diff
@@ -57,17 +58,37 @@ class EventEmitterMixin {
   gin::ObjectTemplateBuilder GetObjectTemplateBuilder(v8::Isolate* isolate) {
     gin::PerIsolateData* data = gin::PerIsolateData::From(isolate);
     auto* wrapper_info = &(static_cast<T*>(this)->kWrapperInfo);
-    v8::Local<v8::FunctionTemplate> constructor =
-        data->DeprecatedGetFunctionTemplate(wrapper_info);
+
+    // DeprecatedWrapperInfo support will be removed as part of
+    // https://github.com/electron/electron/issues/47922
+    constexpr bool is_deprecated_wrapper =
+        std::is_same_v<decltype(wrapper_info), gin::DeprecatedWrapperInfo*>;
+
+    v8::Local<v8::FunctionTemplate> constructor;
+    if constexpr (is_deprecated_wrapper) {
+      constructor = data->DeprecatedGetFunctionTemplate(wrapper_info);
+    } else {
+      constructor = data->GetFunctionTemplate(wrapper_info);
+    }
+
+    const char* class_name = "";
+    if constexpr (is_deprecated_wrapper) {
+      class_name = static_cast<T*>(this)->GetTypeName();
+    } else {
+      class_name = static_cast<T*>(this)->GetHumanReadableName();
+    }
```

Comment:
We are using `GetHumanReadableName` specifically to show up in snapshots and have prefixed them with `Electron / <ClassName>` to keep them unique. It cannot be used directly as class constructor names. We can have a `GetClassName` member that can be used here instead, an example is https://github.com/electron/electron/blob/1f322ebce3cccc65b32b653ea15bc1511a0b5993/shell/browser/api/electron_api_session.h#L92